### ISSUE TITLE
java.lang.ClassCastException: clojure.lang.PersistentArrayMap cannot be cast to java.util.concurrent.Future

### ISSUE THREAD

**Issue by zafartm** [luminus-framework/conman#49]
I was using conman version 0.6.2 for few months in my project. Today upgraded to latest 0.7.3 and started getting this;
java.lang.ClassCastException: clojure.lang.PersistentArrayMap cannot be cast to java.util.concurrent.Future
        at clojure.core$deref_future.invokeStatic(core.clj:2206) ~[clojure-1.8.0.jar:na]
        at clojure.core$deref.invokeStatic(core.clj:2228) ~[clojure-1.8.0.jar:na]
        at clojure.core$deref.invoke(core.clj:2214) ~[clojure-1.8.0.jar:na]

Switching to 0.7.2 solved it though.

**Comment by yogthos** [comment]
@dryewo sounds like it's related to this pr https://github.com/luminus-framework/conman/pull/48

**Comment by dryewo** [comment]
@zafartm could you please provide more details? When does it happen? What is the full stack trace?

**Comment by zafartm** [comment]
It starts happening as soon as I start my application with version 0.7.3 and try to do some database read or write.
xx.xx.storage.db is the namespace where all hugsql functions go.
Full stack trace is here;
```
java.lang.ClassCastException: clojure.lang.PersistentArrayMap cannot be cast to java.util.concurrent.Future
        at clojure.core$deref_future.invokeStatic(core.clj:2206)
        at clojure.core$deref.invokeStatic(core.clj:2228)
        at clojure.core$deref.invoke(core.clj:2214)
        at xx.xx.storage.db$eval8653$f__8163__auto____8682.invoke(db.clj:34)
        at xx.xx.services.users_helper$build_identity.invokeStatic(users_helper.clj:438)
        at xx.xx.services.users_helper$build_identity.invoke(users_helper.clj:434)
        at xx_services.routes.sso$identity_from_ssoassertion.invokeStatic(sso.clj:232)
        at xx_services.routes.sso$identity_from_ssoassertion.invoke(sso.clj:227)
        at xx_services.routes.sso$handle_sso_callback.invokeStatic(sso.clj:248)
        at xx_services.routes.sso$handle_sso_callback.invoke(sso.clj:236)
        at xx_services.routes.sso$fn__19080.invokeStatic(sso.clj:353)
        at xx_services.routes.sso$fn__19080.invoke(sso.clj:353)
        at compojure.core$wrap_response$fn__2258.invoke(core.clj:158)
        at compojure.core$wrap_route_middleware$fn__2242.invoke(core.clj:128)
        at compojure.core$wrap_route_info$fn__2247.invoke(core.clj:137)
        at compojure.core$wrap_route_matches$fn__2251.invoke(core.clj:146)
        at compojure.core$routing$fn__2266.invoke(core.clj:185)
        at clojure.core$some.invokeStatic(core.clj:2592)
        at clojure.core$some.invoke(core.clj:2583)
        at compojure.core$routing.invokeStatic(core.clj:185)
        at compojure.core$routing.doInvoke(core.clj:182)
        at clojure.lang.RestFn.applyTo(RestFn.java:139)
        at clojure.core$apply.invokeStatic(core.clj:648)
        at clojure.core$apply.invoke(core.clj:641)
        at compojure.core$routes$fn__2270.invoke(core.clj:192)
        at clojure.lang.Var.invoke(Var.java:379)
        at compojure.core$routing$fn__2266.invoke(core.clj:185)
        at clojure.core$some.invokeStatic(core.clj:2592)
        at clojure.core$some.invoke(core.clj:2583)
        at compojure.core$routing.invokeStatic(core.clj:185)
        at compojure.core$routing.doInvoke(core.clj:182)
        at clojure.lang.RestFn.applyTo(RestFn.java:139)
        at clojure.core$apply.invokeStatic(core.clj:648)
        at clojure.core$apply.invoke(core.clj:641)
        at compojure.core$routes$fn__2270.invoke(core.clj:192)
        at clojure.lang.Var.invoke(Var.java:379)
        at ring.middleware.reload$wrap_reload$fn__34590.invoke(reload.clj:39)
        at selmer.middleware$wrap_error_page$fn__34603.invoke(middleware.clj:9)
        at prone.middleware$wrap_exceptions$fn__34801.invoke(middleware.clj:126)
        at xx_services.logging_middleware$wrap_application_logging$fn__34898.invoke(logging_middleware.clj:147)
        at xx_services.auths_backend$wrap_session_activity$fn__24791.invoke(auths_backend.clj:54)
        at buddy.auth.middleware$wrap_authorization$fn__24459.invoke(middleware.clj:110)
        at buddy.auth.middleware$wrap_authentication$fn__24452.invoke(middleware.clj:59)
        at ring.middleware.session$wrap_session$fn__24693.invoke(session.clj:108)
        at ring.middleware.keyword_params$wrap_keyword_params$fn__20851.invoke(keyword_params.clj:36)
        at ring.middleware.nested_params$wrap_nested_params$fn__20899.invoke(nested_params.clj:89)
        at ring.middleware.multipart_params$wrap_multipart_params$fn__28597.invoke(multipart_params.clj:172)
        at ring.middleware.params$wrap_params$fn__20973.invoke(params.clj:67)
        at ring.middleware.cookies$wrap_cookies$fn__24572.invoke(cookies.clj:175)
        at ring.middleware.absolute_redirects$wrap_absolute_redirects$fn__35163.invoke(absolute_redirects.clj:47)
        at ring.middleware.resource$wrap_resource$fn__35092.invoke(resource.clj:37)
        at ring.middleware.content_type$wrap_content_type$fn__23596.invoke(content_type.clj:34)
        at ring.middleware.default_charset$wrap_default_charset$fn__35135.invoke(default_charset.clj:31)
        at ring.middleware.not_modified$wrap_not_modified$fn__23626.invoke(not_modified.clj:53)
        at ring.middleware.x_headers$wrap_x_header$fn__34962.invoke(x_headers.clj:22)
        at ring.middleware.x_headers$wrap_x_header$fn__34962.invoke(x_headers.clj:22)
        at ring.middleware.x_headers$wrap_x_header$fn__34962.invoke(x_headers.clj:22)
        at immutant.web.internal.undertow$create_http_handler$reify__38474.handleRequest(undertow.clj:239)
        at org.projectodd.wunderboss.web.undertow.async.websocket.UndertowWebsocket$2.handleRequest(UndertowWebsocket.java:109)
        at io.undertow.server.session.SessionAttachmentHandler.handleRequest(SessionAttachmentHandler.java:68)
        at io.undertow.server.Connectors.executeRootHandler(Connectors.java:211)
        at io.undertow.server.HttpServerExchange$1.run(HttpServerExchange.java:809)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
        at java.lang.Thread.run(Unknown Source)
```

**Comment by dryewo** [comment]
hmm, please tell me more
1. Do you use `:cljc` mode in mount (it's off by default, enabled by calling `(mount/in-cljc-mode)`)?
2. Does this exception happen inside a transaction? If yes, do you use `conman/transaction` or `clojure.java.jdbc/with-db-transaction`? Do you nest transactions?
3. Do you use `conman/bind-connection` and then use functions without explicitly giving them the db connection (`*db*`)? 
4. Does this specific exception occur when you call one of the hugsql's functions or just `clojure.java.jdbc/query`? If hugsql, then how many parameters does this call use have?

**Comment by zafartm** [comment]
- no explicit :cljc mode.
- no explicit transactions
- yes I use bind-connection then call functions defined in .sql files. without explicitly giving them the db connection.
- yes the exception occurs when I call hugsql function. I do not use clojure.java.jdbc/query.
- all my hugsql functions take a map in parameter. (or no paramater at all)

**Comment by zafartm** [comment]
And I do all this bcz this is the only way I know so far. If there are alternates, please let me know (by sharing some docs etc.) I will be grateful. @dryewo

**Comment by dryewo** [comment]
I reproduced, looking into it.

**Comment by dryewo** [comment]
@zafartm This should solve the problem: #50

**Comment by yogthos** [comment]
thanks for the quick fix, I pushed out `0.7.4` @zafartm let us know if this fixes the issue for you

**Comment by zafartm** [comment]
I confirm. Version 0.7.4 fixes the issue.
